### PR TITLE
dash/dataGrid-editableOptions

### PR DESCRIPTION
**Pull request by karolkolodziej**
Added `editableOptions` for the DataGrid component.

- [x] Styling
- [x] Tests

### CODE REVIEW

File: test/cypress/dashboards/integration/Components/DataGrid/DataGridComponent.cy.js [hunk 1]
```diff
@@ -110,4 +110,59 @@ describe('layout resize on window changes', () => {
...
+            .find('input[type="checkbox"]')
+            .should('not.be.checked');
+    });
+
+    it('Changing options in the sidebar should update the grid.', () => {
+        // Act
+        cy.toggleEditMode();
+        cy.openCellEditSidebar('#dashboard-col-1');
+
+        //Assert
```
Comment (sebastianbochan) [inline test/cypress/dashboards/integration/Components/DataGrid/DataGridComponent.cy.js:158]: space

**Review by sebastianbochan — APPROVED** [review]

**Review by jomar-honsi — APPROVED** [review]

**Review by ddragula — CHANGES_REQUESTED** [review]
It looks great! But I found one issue: not all changed options are reverted after editing cancel. Could you investigate it?

File: ts/Dashboards/Components/DataGridComponent/DataGridComponentDefaults.ts [hunk 2]
```diff
@@ -40,11 +40,66 @@ const DataGridComponentDefaults: Globals.DeepPartial<Options> = {
...
+        }, {
+            name: 'title',
+            propertyPath: ['title'],
+            type: 'input'
+        }, {
+            name: 'caption',
+            propertyPath: ['caption'],
+            type: 'input'
+        }, {
+            name: 'Grid caption',
```
Comment (stitot) [inline ts/Dashboards/Components/DataGridComponent/DataGridComponentDefaults.ts:57]:
```suggestion
            name: 'DataGrid caption/title',
```

File: ts/Dashboards/Components/DataGridComponent/DataGridComponentDefaults.ts [hunk 3]
```diff
@@ -40,11 +40,66 @@ const DataGridComponentDefaults: Globals.DeepPartial<Options> = {
...
+            propertyPath: ['dataGridOptions', 'settings', 'caption', 'text'],
+            type: 'input'
+        }, {
+            name: 'Editable table',
```
Comment (stitot) [inline ts/Dashboards/Components/DataGridComponent/DataGridComponentDefaults.ts:61]:
```suggestion
            name: 'Editable DataGrid',
```

File: ts/Dashboards/Components/DataGridComponent/DataGridComponentDefaults.ts [hunk 4]
```diff
@@ -40,11 +40,66 @@ const DataGridComponentDefaults: Globals.DeepPartial<Options> = {
...
+            propertyPath:
+                ['dataGridOptions', 'settings', 'columns', 'distribution'],
+            type: 'select',
+            selectOptions: [{
+                name: 'full'
+            }, {
+                name: 'fixed'
+            }]
+        }, {
+            name: 'Text truncation',
```
Comment (stitot) [inline ts/Dashboards/Components/DataGridComponent/DataGridComponentDefaults.ts:96]:
```suggestion
            name: 'Cell text truncation',
```

**Review by stitot — CHANGES_REQUESTED** [review]
Looks really good!!

In addition to inline comments: Spec probably says otherwise, but when looking at it now I believe all DataGrid related options should be moved into a separate "panel" called DataGrid options".  Sorry about that...

**Comment by karolkolodziej** [comment]
@stitot Please check if the nested options are fine for you. Don't look at the styling for now.
The current implementation doesn't allow to have standalone options in the first parent...

**Comment by stitot** [comment]
> @stitot Please check if the nested options are fine for you. Don't look at the styling for now. The current implementation doesn't allow to have standalone options in the first parent...

Looks good (if not looking at styling ;-), but some changes:

1. DataGrid options -> General -> Caption is caption for the component, so belongs in the root after Title
2. With this grouping you can rename DataGrid options -> General -> "DataGrid Caption/Title" to just "Caption/title"
3. Place DataGrid options -> General -> Columns distribution as the third item, so that all radios are at the bottom.

**Comment by stitot** [comment]
One more:

To be consistent with options we have for charts we should also add "DataGrid class name" and "DataGrid ID" to the root here. If not users will not be able to add DataGrid with different styling, if needed.

**Comment by karolkolodziej** [comment]
@ddragula I haven't found this issue. Do you have any steps on how to recreate it?

File: ts/Dashboards/Components/DataGridComponent/DataGridComponentDefaults.ts [hunk 5]
```diff
@@ -40,11 +40,99 @@ const DataGridComponentDefaults: Globals.DeepPartial<Options> = {
...
+                ]
+            }]
+        }, {
+            name: 'DataGrid class name',
+            propertyPath: ['dataGridClassName'],
+            type: 'input'
+        }, {
+            name: 'DataGrid ID',
+            propertyPath: ['dataGridID'],
+            type: 'input'
```
Comment (stitot) [inline ts/Dashboards/Components/DataGridComponent/DataGridComponentDefaults.ts:133]: These are not updated on unfocus and save, when checking /cypress/component-datagrid.

I my previous review I wrote "we should also add "DataGrid class name" and "DataGrid ID" to the root here". The placement is to be consistent with other components, but I've added a separate task (https://app.asana.com/0/1203234853779898/1208086790358623) for a cleanup here. Let's look at all components together later on.

**Review by stitot — CHANGES_REQUESTED** [review]
Looks good! Just one comment.

**Review by stitot — APPROVED** [review]
Nice!

**Review by ddragula — APPROVED** [review]

**Pull request merged**